network/tcp.go

```diff
@@ -72,27 +72,26 @@ func NewTCPConn(addr Address) (conn *TCPConn, err error) {
 }
 
 // Receive get the bytes from the connection then decodes the buffer.
-// It returns the Packet with the Msg field decoded
-// or EmptyApplicationPacket and an error if something wrong happened.
-func (c *TCPConn) Receive() (nm Packet, e error) {
+// It returns the Envelope containing the message,
+// or EmptyEnvelope and an error if something wrong happened.
+func (c *TCPConn) Receive() (nm Envelope, e error) {
```

Comment:
Well there is no "why", we just did not do it like that ;)
Feel free to issue that or add to existing network issue.